iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestHiveIcebergStorageHandlerNoScan.java

```diff
@@ -155,6 +155,93 @@ public void after() throws Exception {
     HiveIcebergStorageHandlerTestUtils.close(shell);
   }
 
+  @Test
+  public void testPartitionEvolution() {
```

Comment:
What happens when I try to alter an HBase table with partition spec, or a normal Hive (non-Iceberg) table?

Reply:
A Semantic exception will be raised. I created a q test to cover this scenario.